modules/adotBidAdapter.js

```diff
@@ -0,0 +1,636 @@
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {BANNER, NATIVE, VIDEO} from '../src/mediaTypes.js';
+
+const ADAPTER_VERSION = 'v1.0.0';
+const BID_METHOD = 'POST';
+const BIDDER_URL = 'https://dsp.adotmob.com/headerbidding/bidrequest';
+const SUPPORTED_VIDEO_CONTEXTS = ['instream', 'outstream'];
+const SUPPORTED_INSTREAM_CONTEXTS = ['pre-roll', 'mid-roll', 'post-roll'];
+const NATIVE_PLACEMENTS = {
+  title: {id: 1, name: 'title'},
+  icon: {id: 2, type: 1, name: 'img'},
+  image: {id: 3, type: 3, name: 'img'},
+  sponsoredBy: {id: 4, name: 'data', type: 1},
+  body: {id: 5, name: 'data', type: 2},
+  cta: {id: 6, type: 12, name: 'data'}
+};
+const NATIVE_ID_MAPPING = {1: 'title', 2: 'icon', 3: 'image', 4: 'sponsoredBy', 5: 'body', 6: 'cta'};
+const SUPPORTED_VIDEO_MIMES = ['video/mp4'];
+const DOMAIN_REGEX = new RegExp('//([^/]*)');
+const FIRST_PRICE = 1;
+const BID_SUPPORTED_MEDIA_TYPES = ['banner', 'video', 'native'];
+const TTL = 10;
+const NET_REVENUE = true;
+// eslint-disable-next-line no-template-curly-in-string
+const AUCTION_PRICE = '${AUCTION_PRICE}';
+const OUTSTREAM_VIDEO_PLAYER_URL = 'http://adserver.adotmob.com/video/player.min.js';
+
+function isObject(value) {
+  return (value !== null) && (typeof (value) === 'object') && !isArray(value);
+}
+
+function isArray(value) {
+  return Array.isArray(value);
+}
+
+function isString(value) {
+  return (typeof (value) === 'string');
+}
+
+function isNumber(value) {
+  return (typeof (value) === 'number');
+}
+
+function isBoolean(value) {
+  return (typeof (value) === 'boolean');
+}
+
+function isNone(value) {
+  return (value === null) || (value === undefined);
+}
+
+function groupBy(values, key) {
+  const groups = values.reduce((acc, value) => {
+    const groupId = value[key];
+
+    if (!acc[groupId]) acc[groupId] = [];
+    acc[groupId].push(value);
+
+    return acc;
+  }, {});
+
+  return Object
+    .entries(groups)
+    .map(([id, values]) => ({id, key, values}));
+}
+
+function validateMediaTypes(mediaTypes, allowedMediaTypes) {
+  if (!isObject(mediaTypes)) return false;
+  if (!allowedMediaTypes.some(mediaType => mediaType in mediaTypes)) return false;
+
+  if (isBanner(mediaTypes)) {
+    if (!validateBanner(mediaTypes.banner)) return false;
+  }
+
+  if (isVideo(mediaTypes)) {
+    if (!validateVideo(mediaTypes.video)) return false;
+  }
+
+  return true;
+}
+
+function isBanner(mediaTypes) {
+  return isObject(mediaTypes) && isObject(mediaTypes.banner);
+}
+
+function isVideo(mediaTypes) {
+  return isObject(mediaTypes) && isObject(mediaTypes.video);
+}
+
+function validateBanner(banner) {
+  return isObject(banner) &&
+    isArray(banner.sizes) &&
+    (banner.sizes.length > 0) &&
+    banner.sizes.every(validateMediaSizes);
+}
+
+function validateVideo(video) {
+  if (!isObject(video)) return false;
+  if (!isString(video.context)) return false;
+  if (!SUPPORTED_VIDEO_CONTEXTS.includes(video.context)) return false;
```

Comment:
The `includes` here isn't supported by IE11.  Is there an alternative you can use?